repose-aggregator/components/filters/body-extractor-to-header/src/main/resources/META-INF/schema/config/body-extractor-to-header.xsd

```diff
@@ -0,0 +1,115 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+  ~ Repose
+  ~ _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+  ~ Copyright (C) 2010 - 2015 Rackspace US, Inc.
+  ~ _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~      http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~ =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+  -->
+
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
+           xmlns:html="http://www.w3.org/1999/xhtml"
+           xmlns="http://docs.openrepose.org/repose/body-extractor-to-header/v1.0"
+           targetNamespace="http://docs.openrepose.org/repose/body-extractor-to-header/v1.0"
+           elementFormDefault="qualified"
+           attributeFormDefault="unqualified">
+    <xs:element name="body-extractor-to-header" type="BodyExtractorToHeaderConfig"/>
+
+    <xs:complexType name="BodyExtractorToHeaderConfig">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>The root config type for the Body Extractor to Header filter configuration file.</html:p>
+            </xs:documentation>
+        </xs:annotation>
+        <xs:sequence>
+            <xs:element name="extraction" type="Extractor" minOccurs="1" maxOccurs="unbounded"/>
+        </xs:sequence>
+    </xs:complexType>
+
+    <xs:complexType name="Extractor">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>Config type for a Body extraction to a header value.</html:p>
+            </xs:documentation>
+        </xs:annotation>
+        <xs:attribute name="header" type="xs:string" use="required">
+            <xs:annotation>
+                <xs:documentation>
+                    <html:p>The name of the header to put the extracted value into.</html:p>
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="jsonpath" type="xs:string" use="required">
+            <xs:annotation>
+                <xs:documentation>
+                    <html:p>
+                        The JPath to be applied to the Body in order to extract the desired value.
+                    </html:p>
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="default" type="xs:string" use="optional">
```

Comment:
I don't love naming this attribute "default". I worry that name is overloaded with meaning. Also, while it behaves much like "nullValue", there isn't a clear link between the two, which may be helpful.

"onNull" and "onMiss" were proposed last week. I would also throw "ifNull" and "ifAbsent" into the mix.

I'll let @adrianjgeorge give his opinion on this so that it doesn't come back later.